File: src/main/java/com/lykke/matching/engine/daos/AssetPair.java
package com.lykke.matching.engine.daos;

import com.microsoft.azure.storage.table.TableServiceEntity;

public class AssetPair extends TableServiceEntity {

    public static String ASSET_PAIR = "AssetPair";

    String baseAssetId;
    String quotingAssetId;
    int accuracy;
    int invertedAccuracy;

    public AssetPair() {
    }

    public AssetPair(String baseAssetId, String quotingAssetId, int accuracy, int invertedAccuracy) {
        super(ASSET_PAIR, baseAssetId + quotingAssetId);
        this.baseAssetId = baseAssetId;
        this.quotingAssetId = quotingAssetId;
        this.accuracy = accuracy;
        this.invertedAccuracy = invertedAccuracy;
    }

    public String getBaseAssetId() {
        return baseAssetId;
    }

    public void setBaseAssetId(String baseAssetId) {
        this.baseAssetId = baseAssetId;
    }

    public String getQuotingAssetId() {
        return quotingAssetId;
    }

    public void setQuotingAssetId(String quotingAssetId) {
        this.quotingAssetId = quotingAssetId;
    }

    public String getAssetPairId() {
        return rowKey;
    }

    public int getAccuracy() {
        return accuracy;
    }

    public void setAccuracy(int accuracy) {
        this.accuracy = accuracy;
    }

    public int getInvertedAccuracy() {
        return invertedAccuracy;
    }

    public void setInvertedAccuracy(int invertedAccuracy) {
        this.invertedAccuracy = invertedAccuracy;
    }

    @Override
    public String toString() {
        return "AssetPair(pairId=" + getAssetPairId() + ",baseAssetId=" + baseAssetId + ", quotingAssetId=" + quotingAssetId + ", accuracy=" + accuracy + ", invertedAccuracy=" + invertedAccuracy + ")";
    }
}

File: src/main/kotlin/com/lykke/matching/engine/daos/AssetPair.kt
package com.lykke.matching.engine.daos

import java.math.BigDecimal

class AssetPair(
        val assetPairId: String,
        val baseAssetId: String,
        val quotingAssetId: String,
        val accuracy: Int,
        val minVolume: BigDecimal? = null,
        val minInvertedVolume: BigDecimal? = null,
        val maxValue: BigDecimal? = null,
        val midPriceDeviationThreshold: BigDecimal? = null,
        val marketOrderPriceDeviationThreshold: BigDecimal? = null
) {
    override fun toString(): String {
        return "AssetPair(" +
                "assetPairId='$assetPairId', " +
                "baseAssetId='$baseAssetId', " +
                "quotingAssetId='$quotingAssetId', " +
                "accuracy=$accuracy, " +
                "minVolume=$minVolume, " +
                "minInvertedVolume=$minInvertedVolume, " +
                "maxValue=$maxValue, " +
                "midPriceDeviationThreshold=$midPriceDeviationThreshold, " +
                "marketOrderPriceDeviationThreshold=$marketOrderPriceDeviationThreshold"
    }
}